docs/extensions/payment-processors/index.md

```diff
@@ -0,0 +1,190 @@
+# Payment Processors
+
+This page documents payment processing in CiviCRM. The key bit to understand is the payment processor object, a civicrm-specific way of representing and integrating an actual real-life payment processor (e.g. like Paypal, etc.). The object and how it is used (as of CiviCRM 4.4) is in need of some love and refactoring, so this page hopes to provide some help in that direction.
+
+
+## Definitions
+
+Some of these definitions are a bit loose, but should at least help
+clarify the use of these over-used words in the rest of the page.
+
+**Payment Processor Provider** is a commercial entity that will sell you
+one or more payment processor services.
+
+**Payment Processor** is a real-life service or process that enables a
+constituent to give money to the organization that owns the CiviCRM
+instance. Many payment processor service providers enable more than one
+payment processor. Not all payment processors need to be attached to a
+provider (in theory).
+
+**Payment Class** is a php object class that represents a payment using
+a processor, and provides methods and properties that integrate the
+processor's functions into CiviCRM.
+
+**Payment Type** is a CiviCRM-specific classification of payment types -
+an integer representing the 'type' of payment this payment processor
+class supports. Almost all existing payment processors are of type 1,
+for credit cards, but there is some support in code for 2 ('direct
+debit'). The primary use of this type is for generating a base set of
+fields in the user form.
+
+**Billing Mode** is a CiviCRM-specific classification of different types
```

Comment:
Not sure if this is deprecated or not... probably not actually.  Could do with an idea of when to use each type, and note that they need to be specified as numeric values.